File: 2019/day1/Lua/part2.lua
dofile("UsefulStuff.lua")

local function part1(fileName)
	local total = 0
	for fuel in io.lines(Script_path() .. fileName) do
        subTotal = 0
        fuel = tonumber(fuel)
        while(fuel > 0) do
            fuel = math.floor(fuel/3)-2
            if(fuel > 0) then
                subTotal = subTotal + fuel
            end
        end
        total = total + subTotal
	end
	return total
end

UnitTest(part1, "../testcases/test1.txt", 51316)
UnitTest(part1, "../input.txt", 4934153)

local result = part1("../input.txt")
print("Part 2: " .. tostring(result))

File: 2019/day1/Node/part2.js
const { assert } = require('console');
const fs = require('fs');

function part2(filepath) {

    data = fs.readFileSync(filepath, 'UTF8').toString().split('\n');
    sum = 0;
    while (data.length > 0) {
        data = data.map((mass) => Math.floor(mass/3)-2 )
                .filter((fuel) => fuel > 0 );
        data.forEach((validFuel) => {sum+=validFuel});
    }
    return sum
}

assert(part2(__dirname+'/../testcases/test1.txt')==51316)
assert(part2(__dirname+'/../input.txt')==4934153)
console.log("Part 2:",part2(__dirname+'/../input.txt'))

